internal/storage/inmem/snapshot.go

```diff
@@ -0,0 +1,78 @@
+package inmem
+
+import (
+	"github.com/hashicorp/go-memdb"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+// Snapshot obtains a point-in-time snapshot of the store that can later be
+// persisted and restored later.
+func (s *Store) Snapshot() (*Snapshot, error) {
+	tx := s.txn(false)
+
+	iter, err := tx.Get(tableNameResources, indexNameID)
```

Comment:
I don't think we need to because it currently only stores the event index, which doesn't need to be stable between snapshots and restores. It's probably worth putting a comment on the `meta` struct, though, to call out that you shouldn't put anything durable into that table.